addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/dhcp/DHCPPacketListenerServer.java

```diff
@@ -0,0 +1,112 @@
+
+/**
+ * Receives UDP messages and try to parse them as DHCP messages.
+ * First try to listen to the DHCP port 67 and if that failes because of missing access rights,
+ * port 6767 will be opened instead (and the user is required to setup a port forarding).
+ *
+ * @author David Graeff - Initial contribution
+ */
+public class DHCPPacketListenerServer extends Thread {
+    private byte[] buffer = new byte[1024];
+    DatagramSocket dsocket;
+    private DatagramPacket packet = new DatagramPacket(buffer, buffer.length);
+    boolean willbeclosed = false;
+    Logger logger = LoggerFactory.getLogger(DHCPPacketListenerServer.class);
+    private boolean useUnprevilegedPort = false;
+    private final IPRequestReceivedCallback listener;
+
+    DHCPPacketListenerServer(IPRequestReceivedCallback listener) throws SocketException, BindException {
+        this.listener = listener;
+        try {
+            dsocket = new DatagramSocket(null);
+            dsocket.setReuseAddress(true);
+            dsocket.setBroadcast(true);
+            dsocket.bind(new InetSocketAddress(67));
+        } catch (SocketException e) {
+            useUnprevilegedPort = true;
+            dsocket = new DatagramSocket(null);
+            dsocket.setReuseAddress(true);
+            dsocket.setBroadcast(true);
+            dsocket.bind(new InetSocketAddress(6767));
+        }
+    }
+
+    void receivePacket(DHCPPacket request, InetAddress udpRemote)
+            throws BadPacketException, UnknownHostException, IOException {
+        if (request.getOp() != DHCPPacket.BOOTREQUEST) {
+            return; // skipping non BOOTREQUEST message types
+        }
+
+        Byte dhcpMessageType = request.getDHCPMessageType();
+
+        if (dhcpMessageType == null || dhcpMessageType != DHCPPacket.DHCPREQUEST) {
+            return; // skipping non DHCPREQUEST message types
+        }
+
+        InetAddress requestedAddress = request.getRequestedIPAddress();
+        if (requestedAddress == null) {
+            // There is no requested address field. This may be a DHCPREQUEST message to renew
+            // the lease. Let's deduct the IP by the IP/UDP src.
+            requestedAddress = udpRemote;
+            if (requestedAddress == null) {
+                logger.warn("DHO_DHCP_REQUESTED_ADDRESS field is missing");
+                return;
+            }
+        }
+        listener.dhcpRequestReceived(requestedAddress.getHostAddress());
+    }
+
+    @Override
+    public void run() {
+        try {
+            logger.info("DHCP request packet listener online");
+            while (!willbeclosed) {
+                packet.setLength(buffer.length);
+                dsocket.receive(packet);
+                try {
+                    receivePacket(new DHCPPacket(packet), packet.getAddress());
+                } catch (BadPacketException | IOException e) {
+                    return;
+                }
+
+            }
+        } catch (IOException e) {
```

Comment:
Will this catch any exception I assume they will be caught by the inner catch or do they have a different source.